OsmAnd/src/net/osmand/plus/routing/SuggestionsMapsProvider.java

```diff
@@ -0,0 +1,145 @@
+package net.osmand.plus.routing;
+
+import androidx.annotation.NonNull;
+
+import net.osmand.IndexConstants;
+import net.osmand.Location;
+import net.osmand.data.LatLon;
+import net.osmand.map.WorldRegion;
+import net.osmand.plus.OsmandApplication;
+import net.osmand.plus.onlinerouting.EngineParameter;
+import net.osmand.plus.onlinerouting.OnlineRoutingHelper;
+import net.osmand.plus.onlinerouting.engine.EngineType;
+import net.osmand.plus.onlinerouting.engine.OnlineRoutingEngine;
+import net.osmand.util.Algorithms;
+
+import org.json.JSONException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.LinkedHashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static net.osmand.util.MapUtils.calculateMidPoint;
+
+public class SuggestionsMapsProvider {
+
+	private static final int DISTANCE = 20000;
+	RouteCalculationParams params;
+	private boolean pointOnWater;
+
+	public SuggestionsMapsProvider(RouteCalculationParams params) {
+		this.params = params;
+	}
+
+	public boolean isPointOnWater() {
+		return pointOnWater;
+	}
+
+	public boolean checkIfObjectDownloaded(String downloadName) {
+		final String regionName = Algorithms.capitalizeFirstLetterAndLowercase(downloadName)
```

Comment:
Check is not correct, cause it doesn't check actual data if user has other maps for this bbox with different name